tempesta_fw/t/unit/test_http_sticky.c

```diff
@@ -414,54 +415,106 @@ TEST(http_sticky, sending_302_without_preparing)
 	StickyVal sv = {};
 	TfwConn *c = mock.req->conn;
 
+	/*
+	 * The redirect constructor rewrote mock.req->resp, so we have to
+	 * reinitialize mock.resp to be able to parse the response.
+	 * Unpair request and response first to  avoid "Response-Request pairing
+	 * is broken!" warning.
+	 * We don't pair response w/ request in http_sticky_suite_setup(),
+	 * so tfw_http_msg_free() doesn't unpair the messages and we have to do
+	 * this manually.
+	 */
+	tfw_http_msg_unpair((TfwHttpMsg *)mock.req);
+	tfw_http_msg_free((TfwHttpMsg *)mock.resp);
```

Comment:
Rather than freeing a newly created mock response, we may want to use another test suite with a setup function that doesn't create a mock response in the first place. As we have two groups of tests, one which operates on requests, and other which operates on responses, it's natural to have them separated.

Reply:
The fix is to avoid sporadic crashes during unit tests. The whole `sticky` unit test suite is too complicated and hard to maintain that we decided to drop it and replace with functional tests. See tempesta-tech/tempesta-test#74 . So no further enhancement is planned here and only crucial fixes.

Reply:
OK